packages/fxa-auth-server/lib/payments/paypal-client.ts

```diff
@@ -88,6 +93,39 @@ export type DoReferenceTransactionOptions = {
   billingAgreementId: string;
 };
 
+type IpnBaseType = {
+  txn_type: string;
+  [key: string]: any;
+};
+
+export type IpnMerchPmtType = IpnBaseType & {
+  invoice: string;
+  // The billing agreement ID
+  mp_id: string;
+  mc_gross: string;
+  mp_notification?: any;
+  mp_pay_type: 'INSTANT' | 'ANY' | 'ECHECK';
+  // Agreement status, A for Active, I for inactive
+  // I is equivalent to cancelled.
+  mp_status: 'A' | 'I';
+  txn_id: string;
+  txn_type: 'merch_pmt' | 'mp_cancel';
+};
+
+export type IpnMessage = IpnBaseType;
```

Comment:
Couldn't we rename IpnBaseType to IpnMessage and not have this alias?

Reply:
Ah, yes, that would probably be a bit better, I originally had this more complicated and forgot to remove this.